src/main/asciidoc/java/index.adoc

```diff
@@ -0,0 +1,741 @@
+[source,java]
+----
+public WebServerOptions setCertificates(List<String> certificates) {
+  this.certificates = certificates;
+  return this;
+}
+----
+
+Or an _adder_ :
+
+.a multi valued adder
+[source,java]
+----
+public WebServerOptions addCertificate(String certificate) {
+  this.certificates.add(certificate);
+  return this;
+}
+----
+
+Map multi valued properties can be declared via a _setter_.
+
+NOTE: these examples uses a _fluent_ return types for providing a better API, this is not mandatory but
+encouraged
+
+==== Json conversion
+
+Vert.x requires data object to be convertible from json with the json argument constructor. Altough there
+is no strict rules of mapping between the data object properties and the json structure, it is a good thing
+to follow a common mapping for users using json data objects (like in JavaScript shim).
+
+Json properties should be named after properties according to JavaBean conversion rules:
+
+- a single valued property follows the JavaBean convention
+- a multi valued property declared with a list setter follows the same convention
+- a multi valued property declared with an adder must use a singular form and the json property name gets a trailing _s_
+
+In all case, property names are _normalized_, i.e:
+
+- _red_ -> _red_
+- _Red_ -> _red_
+- _URL_ -> _url_
+- _URLFactory_ -> _urlFactory_
+
+==== Jsonifiable data object
+
+When a data object declares a `public JsonObject toJson()` method it can be converted to the json format
+and is said _jsonifiable_. Vert.x API types have restriction in the declared method return types, a jsonifiable
+data object can be used in Vert.x API method return types or handlers because it can be converted to a json
+format, otherwise it is not permitted.
+
+==== Data object converter
+
+Vert.x requires only the conversion from json, but the conversion to json can be useful when the API uses
+data objects as return types. The implementation of the data object / json conversion can be tedious and
+error prone.
+
+The data object / json conversion can be automated based on the rules given above. Vert.x Core allows to
+generate auxilliary classes that implement the conversion logic. The generated converters handle the
+type mapping as well as the json naming convention.
+
+Converters are generated when the data object is annotated with `@DataObject(generateConverter=true)`. The
+generation happens for the data object properties, not for the ancestor properties, unless `inheritConverter`
+is set: `@DataObject(generateConverter=true,inheritConverter=true)`.
+
+The converter is named by appending the `Converter` suffix to the data object class name, e.g,
+`ContactDetails` -> `ContactDetailsConverter`. The generated converter has two static methods:
+
+- `public static void fromJson(JsonObject json, ContactDetails obj)`
+- `public static void toJson(ContactDetails obj, JsonObject json)`
+
+The former should be used in the json constructor, the later in the `toJson` method.
```

Comment:
Maybe you should add an example with a constructor taking the JSON object are argument and the toJson method.